From f0db4a6ba10ab8b94949f01aae9e540847cd99e9 Mon Sep 17 00:00:00 2001
From: Kshitiz Varshney <kshitiz.varshney@nxp.com>
Date: Mon, 9 Jan 2023 16:17:51 +0100
Subject: [PATCH] imx_v8_defconfig: IMA-EVM kernel config

This patch is for applying IMA-EVM specific configs to vanilla kernel.

Signed-off-by: Kshitiz Varshney <kshitiz.varshney@nxp.com>
---
 arch/arm64/configs/imx_v8_defconfig | 32 +++++++++++++++++++++++++++--
 1 file changed, 30 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/configs/imx_v8_defconfig b/arch/arm64/configs/imx_v8_defconfig
index e98cd78c98c7..0a3dc3cf9a2b 100644
--- a/arch/arm64/configs/imx_v8_defconfig
+++ b/arch/arm64/configs/imx_v8_defconfig
@@ -1013,7 +1013,7 @@ CONFIG_ROOT_NFS=y
 CONFIG_9P_FS=y
 CONFIG_NLS_CODEPAGE_437=y
 CONFIG_NLS_ISO8859_1=y
-CONFIG_TRUSTED_KEYS=m
+CONFIG_TRUSTED_KEYS=y
 # CONFIG_TRUSTED_KEYS_TPM is not set
 # CONFIG_TRUSTED_KEYS_TEE is not set
 CONFIG_SECURITY=y
@@ -1062,7 +1062,7 @@ CONFIG_CRYPTO_AES_ARM64_BS=m
 CONFIG_CRYPTO_AES_ARM64_CE_CCM=y
 CONFIG_CRYPTO_CRCT10DIF_ARM64_CE=m
 CONFIG_CRYPTO_DEV_FSL_CAAM_SECVIO=m
-CONFIG_CRYPTO_DEV_FSL_CAAM=m
+CONFIG_CRYPTO_DEV_FSL_CAAM=y
 CONFIG_CRYPTO_DEV_FSL_CAAM_SM_TEST=m
 CONFIG_CRYPTO_DEV_FSL_DPAA2_CAAM=m
 CONFIG_CRYPTO_DEV_CCREE=m
@@ -1094,3 +1094,31 @@ CONFIG_CORESIGHT_STM=m
 CONFIG_CORESIGHT_CPU_DEBUG=m
 CONFIG_CORESIGHT_CTI=m
 CONFIG_MEMTEST=y
+#
+# Security options
+#
+CONFIG_SECURE_KEYS=n
+#CONFIG_ENCRYPTED_KEYS=y
+CONFIG_SECURITYFS=y
+CONFIG_IMA=y
+CONFIG_IMA_MEASURE_PCR_IDX=10
+CONFIG_IMA_NG_TEMPLATE=y
+CONFIG_IMA_DEFAULT_TEMPLATE="ima-ng"
+CONFIG_IMA_DEFAULT_HASH_SHA1=y
+CONFIG_IMA_DEFAULT_HASH="sha1"
+CONFIG_IMA_APPRAISE=y
+CONFIG_IMA_APPRAISE_BOOTPARAM=y
+CONFIG_EVM=y
+CONFIG_EVM_ATTR_FSUUID=y
+CONFIG_EVM_ADD_XATTRS=y
+CONFIG_INTEGRITY=y
+CONFIG_INTEGRITY_SIGNATURE=y
+CONFIG_INTEGRITY_AUDIT=y
+CONFIG_IMA_LSM_RULES=y
+CONFIG_IMA_TRUSTED_KEYRING=y
+CONFIG_EXT4_FS_SECURITY=y
+CONFIG_IMA_QUEUE_EARLY_BOOT_KEYS=y
+
+
+CONFIG_IMX_EL_ENCLAVE=y
+
-- 
2.25.1

